hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/web/DatanodeHttpServer.java

```diff
@@ -144,7 +146,9 @@ public DatanodeHttpServer(final Configuration conf,
     confForCreate.set(FsPermission.UMASK_LABEL, "000");
 
     this.bossGroup = new NioEventLoopGroup();
-    this.workerGroup = new NioEventLoopGroup();
+    int workerThreads = conf.getInt(DFS_DATANODE_NETTY_WORKER_NUM_THREADS_KEY,
+        DFS_DATANODE_NETTY_WORKER_NUM_THREADS_DEFAULT);
+    this.workerGroup = new NioEventLoopGroup(workerThreads);
```

Comment:
Better to use a cached thread pool here. like `this.workerGroup = new NioEventLoopGroup(workerCount, Executors.newCachedThreadPool())`.